File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messageitem/File.java
package forpleuvoir.dhwu1a.core.message.messageitem;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType;

import static forpleuvoir.dhwu1a.core.common.ApiKey.*;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messageitem
 * <p>#class_name File
 * <p>#create_time 2021/6/30 0:29
 */
public class File extends MessageItem {
    /**
     * 文件识别id
     */
    @SerializedName(ID)
    public final String id;
    /**
     * 文件名
     */
    @SerializedName(NAME)
    public final String name;
    /**
     * 文件大小
     */
    @SerializedName(SIZE)
    public final Long size;

    public File(String id, String name, Long size) {
        super(MessageItemType.File);
        this.id = id;
        this.name = name;
        this.size = size;
    }

    @Override
    public String toPlainText() {
        return String.format("文件(%s)", name);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messageitem/File.kt
package forpleuvoir.dhwu1a.core.message.messageitem

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.ID
import forpleuvoir.dhwu1a.core.common.NAME
import forpleuvoir.dhwu1a.core.common.SIZE
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messageitem
 *
 * #class_name File
 *
 * #create_time 2021/6/30 0:29
 */
class File(
    /**
     * 文件识别id
     */
    @field:SerializedName(ID) val id: String,
    /**
     * 文件名
     */
    @field:SerializedName(NAME) val name: String,
    /**
     * 文件大小
     */
    @field:SerializedName(SIZE) val size: Long
) : MessageItem(MessageItemType.File) {
    override fun toPlainText(): String {
        return String.format("文件(%s)", name)
    }
}